### PR TITLE
Fix/tdp 3715 Preparation is created although close the dataset without saving preparation

### PR DESCRIPTION
**Pull request by charlesnguyen**
**Link to the JIRA issue**
e.g. https://jira.talendforge.org/browse/TDP-3715

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [x] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

**(Optional) What is the current behavior?**
(Additional information to the Jira)

**(Optional) What is the new behavior?**
(Additional information to the Jira)

**(Optional) Other information**:

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/249/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/249/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/250/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/250/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/251/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/251/)

File: dataprep-webapp/src/app/components/playground/playground-controller.spec.js [hunk 1]
```diff
@@ -354,6 +354,27 @@ describe('Playground controller', () => {
 		});
 
 		describe('discard preparation', () => {
+			it('should call startLoader', inject((PlaygroundService) => {
+				// given
+				spyOn(PlaygroundService, 'startLoader');
```
Comment (amaalej) [inline dataprep-webapp/src/app/components/playground/playground-controller.spec.js]: need a new line here, no?

File: dataprep-webapp/src/app/components/playground/playground-controller.spec.js [hunk 2]
```diff
@@ -354,6 +354,27 @@ describe('Playground controller', () => {
...
+				// when
+				ctrl.discardSaveOnClose();
+
+				// then
+				expect(PlaygroundService.startLoader).toHaveBeenCalled();
+			}));
+
+			it('should call stopLoader', inject((PlaygroundService) => {
+				// given
+				spyOn(PlaygroundService, 'stopLoader');
```
Comment (amaalej) [inline dataprep-webapp/src/app/components/playground/playground-controller.spec.js]: need a new line here, no?

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/253/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/253/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/255/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/255/)

**Pull request merged**